jbpm/jbpm-flow/src/main/java/org/jbpm/process/core/context/variable/Variable.java

```diff
@@ -64,6 +69,27 @@ public String getName() {
 
     public void setName(final String name) {
         this.name = name;
+        this.sanitizedName = sanitizeIdentifier(name);
+    }
+
+    public String getSanitizedName() {
+        return sanitizedName;
+    }
+
+    /**
+     * Return a valid unique Java identifier based on the given @param name. It consider valid characters and
+     * reserved words.
+     * In case the input is valid it is returned itself otherwise a unique valid identifier is generated.
+     * @param name the input
+     * @return the output valid Java identifier
+     */
+    private static String sanitizeIdentifier(String name) {
+        return Optional.ofNullable(name)
+                        .filter(SourceVersion::isName)
+                        .orElseGet(() -> Optional.ofNullable(StringUtils.extractFirstIdentifier(name, 0))
+                                .filter(s -> !StringUtils.isEmpty(s))
+                                .filter(SourceVersion::isName)
+                                .orElseGet(() -> "attribute" + StringUtils.generateUUID()));
```

Comment:
I think we can simplify this by just prepending a "$" sign or a mix of "$" sign and some other stuff. This will keep it readable and make it safe. 

```suggestion
       // prepend v$ in front of the variable name to prevent clashing with reserved keywords
        return String.format("v$%s", name);
```

Reply:
done.